docs/api.md

```diff
@@ -0,0 +1,368 @@
+
+e.g. `https://api.dyd.exchange/v1/orders`
+
+## Orderbook
+
+### Introduction
+
+The following API endpoints allow for submitting and retrieving orders from the dYdX orderbook.
+This orderbook is what's frequently referred to as a "Matching Model" orderbook. This means that
+all orders are submitted to the blockchain by dYdX itself You do not need to provide gas fees
+or send on-chain transactions yourself. This is especially useful for traders and market makers who
+wish to be able to quickly cancel their orders without waiting for a transaction to be mined.
+
+In order to submit an order, you (the maker) must first create a JSON object that specifies the
+details of your order. Once you create this object you must sign it with your Ethereum private key,
+and put the result in the `typedSignature` field. Note: The `typedSignature` is omitted before
+signing, and added only after signing the message.
+
+After this is done, the order is ready to be submitted to the API.
+
+__Order fields__
+
+|Field Name|JSON type|Description|
+|----------|---------|-----------|
+|makerMarket|string|The Solo [market](https://docs.dydx.exchange/#/overview?id=markets) of the Maker amount|
+|takerMarket|string|The Solo [market](https://docs.dydx.exchange/#/overview?id=markets) of the Taker amount|
+|makerAmount|string|The amount of token the Maker is offering in base units|
+|takerAmount|string|The amount of token the Maker is requesting from the taker base units|
+|makerAccountOwner|string|The Ethereum address of the Maker.|
+|takerAccountOwner|string|The Ethereum address of the Taker.|
+|makerAccountNumber|string|The Solo [account number](https://docs.dydx.exchange/#/overview?id=markets) of the Maker|
+|takerAccountNumber|string|The Solo [account number](https://docs.dydx.exchange/#/overview?id=markets) of the Taker|
+|expiration|string|The time in unix seconds at which this order will be expired and can no longer be filled.|
+|salt|string|A random number to make the orderHash unique.|
+|typedSignature|string|The signature of the order.|
+
+Example:
+```JSON
+{
+    "makerMarket": "0",
+    "takerMarket": "1",
+    "makerAmount": "10000000000",
+    "takerAmount": "20000000000",
+    "makerAccountOwner": "0x3E5e9111Ae8eB78Fe1CC3bb8915d5D461F3Ef9A9",
+    "makerAccountNumber": "111",
+    "takerAccountOwner": "0x28a8746e75304c0780E011BEd21C72cD78cd535E",
+    "takerAccountNumber": "222",
+    "expiration": "4294967295",
+    "salt": "100",
+    "typedSignature": "0xd9c006cf9066e89c2e75de72604751f63985f173ca3c69b195f1f5f445289a1f2229c0475949858522c821190c5f1ec387f31712bd21f6ac31e4510d5711c2681f00"
+  },
+};
+```
+
+### POST /v1/dex/orders
+
+Description:
+Post a new order to the orderbook.
+
+Headers:
+```
+Content-Type: application/json
+```
+
+Request Body:
+
+|Field Name|JSON type|Description|
+|----------|---------|-----------|
+|order|Object|A valid signed order JSON object|
+|fillOrKill|boolean|Whether or not this order is a market order.|
```

Comment:
A better description might be something like: Whether the order should be canceled if it cannot be immediately filled

(having a fill or kill limit order also makes sense)